src/select-menu/src/OptionsList.js

```diff
@@ -201,6 +204,9 @@ export default class OptionsList extends PureComponent {
 
   handleSelect = item => {
     this.props.onSelect(item)
+    if (!this.props.isMultiSelect && this.props.closeOnSelect) {
```

Comment:
I realized that hitting the `Enter` key has [a similar behavior](https://github.com/segmentio/evergreen/blob/master/src/select-menu/src/OptionsList.js#L190-L192) (closing the menu). It makes me wonder if we should also have the check for `multiSelect` and `closeOnSelect` in the `handleEnter` method (you can always close the menu by hitting `ESC` in that case). Doing this would be a change of contract in the interface. I'm not sure if we want to do that now.

@mshwery - would love some thoughts!

Reply:
I think we could save that for a separate PR if the current behavior is problematic. I think it's fine to leave alone for now TBH, though I agree with your idea!